maven-extension/src/main/java/io/opentelemetry/maven/OpenTelemetrySdkService.java

```diff
@@ -89,52 +82,39 @@ public synchronized void dispose() {
   @Override
   public void initialize() throws InitializationException {
     logger.debug("OpenTelemetry: initialize OpenTelemetrySdkService...");
-    if (StringUtils.isBlank(
-        OtelUtils.getSystemPropertyOrEnvironmentVariable(
-            "otel.exporter.otlp.endpoint", "OTEL_EXPORTER_OTLP_ENDPOINT", null))) {
-      logger.debug(
-          "OpenTelemetry: No -Dotel.exporter.otlp.endpoint property or OTEL_EXPORTER_OTLP_ENDPOINT "
-              + "environment variable found, use a NOOP OpenTelemetry SDK");
-    } else {
-      {
-        // Don't use a {@code io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider} to inject
-        // Maven runtime attributes due to a classloading issue when loading the Maven OpenTelemetry
-        // extension as a pom.xml {@code <extension>}.
-        String initialCommaSeparatedAttributes =
-            OtelUtils.getSystemPropertyOrEnvironmentVariable(
-                "otel.resource.attributes", "OTEL_RESOURCE_ATTRIBUTES", "");
-        Map<String, String> attributes =
-            OtelUtils.getCommaSeparatedMap(initialCommaSeparatedAttributes);
-
-        // service.name
-        String serviceName =
-            OtelUtils.getSystemPropertyOrEnvironmentVariable(
-                "otel.service.name", "OTEL_SERVICE_NAME", null);
-
-        if (!attributes.containsKey(ResourceAttributes.SERVICE_NAME.getKey())
-            && StringUtils.isBlank(serviceName)) {
-          // service.name is not defined in passed configuration, we define it
-          attributes.put(
-              ResourceAttributes.SERVICE_NAME.getKey(),
-              MavenOtelSemanticAttributes.ServiceNameValues.SERVICE_NAME_VALUE);
-        }
-
-        // service.version
-        final String mavenVersion = this.runtimeInformation.getMavenVersion();
-        if (!attributes.containsKey(ResourceAttributes.SERVICE_VERSION.getKey())) {
-          attributes.put(ResourceAttributes.SERVICE_VERSION.getKey(), mavenVersion);
-        }
-
-        String newCommaSeparatedAttributes = OtelUtils.getCommaSeparatedString(attributes);
-        logger.debug(
-            "OpenTelemetry: Initial resource attributes: {}", initialCommaSeparatedAttributes);
-        logger.debug("OpenTelemetry: Use resource attributes: {}", newCommaSeparatedAttributes);
-        System.setProperty("otel.resource.attributes", newCommaSeparatedAttributes);
-      }
-
-      this.openTelemetrySdk = OpenTelemetrySdkAutoConfiguration.initialize(false);
-      this.openTelemetry = this.openTelemetrySdk;
-    }
+    AutoConfiguredOpenTelemetrySdkBuilder autoConfiguredSdkBuilder =
+        AutoConfiguredOpenTelemetrySdk.builder();
+
+    // SDK CONFIGURATION PROPERTIES
+    autoConfiguredSdkBuilder.addPropertiesSupplier(
+        () -> {
+          Map<String, String> properties = new HashMap<>();
+          if ((OtelUtils.getSysPropOrEnvVar("otel.traces.exporter") == null)
```

Comment:
I mentioned it in another comment but how about just default to none regardless of endpoint variable? I think it's reasonable enough to export to localhost, and going from zero to two properties for a remote endpoint isn't that much harder than going from zero to one.

It seems simpler to document, just
The default value for exporter is none
The default value for endpoint is localhost:4317

Instead of

The default value for exporters is none, unless endpoint is explicitly passed in which case it is OTLP.
And endpoint effectively doesn't have a default value.

Former just seems simpler to me, and also fits better with non-otlp exporters which have their own endpoint variables.

Reply:
Ah I didn't mean changing the default of the OTel SDK itself (which is defined by the OTel spec) but just here, removing this `if` statement. We have a general stance of avoiding having default values of two variables associated with each other

https://github.com/open-telemetry/opentelemetry-specification/pull/1895#issuecomment-959861222

Which I think we have here. I don't see the usability drop to be significant enough if a user needs to set both `OTEL_TRACES_EXPORTER` and `OTEL_EXPORTER_OTLP_ENDPOINT` to send to a remote backend to not follow this general policy. Is it ok?

Reply:
I've been trying to figure out from the history but can't seem to.

Is this comment still applicable @anuraaga or was it resolved @cyrille-leclerc?

Happy to review/approve when it's clear

Reply:
Think this comment is still applicable. @cyrille-leclerc what do you think?

Reply:
Sorry for the delay, I was not sure to understand. 

Do you mean that we would change the default behaviour of the "OpenTelemetry SDK Autoconfigure" so that the default value for `OTEL_TRACES_EXPORTER` who no longer be `otlp` but be `none` (official doc [here](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md#exporters))?

This change would make `OTEL_TRACES_EXPORTER` behave the same way `OTEL_METRICS_EXPORTER` behave, correct?

I have few questions please:
* Is it correct that on the Otel Java agent, `OTEL_TRACES_EXPORTER` defaults to `otlp` when `OTEL_METRICS_EXPORTER` defaults to `none` making many users missing Java metrics not catching this subtle nuance of default value? 
* Would this be a breaking change for Otel Java Agent with many existing deployments not defining any of `OTEL_TRACES_EXPORTER` and `OTEL_EXPORTER_OTLP_ENDPOINT`, these deployments would suddenly loose tracing
* The duality `OTEL_TRACES_EXPORTER` / `OTEL_METRICS_EXPORTER` is probably becoming verbose boilerplate with the rise of Otel metrics, including the rising support of metrics in observability backends (e.g. Jaeger is adding support for metrics via PromQL). This verbosity will even increase with the introduction of `OTEL_LOGS_EXPORTER` soon. Would it make sense to introduce a generic `OTEL_EXPORTER` so that the Java SDK exports all the signals it collects to the specified exporter?

Reply:
Sorry if I wasn't clear - I don't mean to change the behavior of SDK autoconfigure (specifically, code in the opentelemetry-java repo) but just to change the defaults here. I'd like to remove handling of `endpoint` here so that

- No settings: No export
- OTEL_EXPORTER_OTLP_ENDPOINT=opentelemetry.io: No export
- OTEL_TRACES_EXPORTER=otlp: Exports to default localhost address
- OTEL_TRACES_EXPORTER=otlp OTEL_EXPORTER_OTLP_ENDPOINT=opentelemetry.io: Exports to opentelemetry.io

I know you are trying to make it simpler to use a remote endpoint, but I believe this provides a more consistent experience with our normal conventions, and don't believe the user experience difference for users of a remote endpoint to be negative enough in practice.

Notably, one issue I find with the current behavior is that
OTEL_TRACES_EXPORTER has a default value of `none`. Yet these two sets have different behavior

- OTEL_EXPORTER_OTLP_ENDPONT=opentelemetry.io: Exports
- OTEL_TRACES_EXPORTER=none OTEL_EXPORTER_OTLP_ENDPOINT=opentelemetry.io: Doesn't export

The behavior shouldn't be different since the default value of OTEL_TRACES_EXPORTER is `none` - we've just added a confusing caveat that the default is different based on the presence of another variable, something we would like to avoid.

This seems like an inconsistent experience. Does it make sense? I can add a commit directly to the PR if things are still unclear, let me know.

> Would it make sense to introduce a generic OTEL_EXPORTER so that the Java SDK exports all the signals it collects to the specified exporter?

This is an interesting topic - it doesn't affect OTel "normally" since exporters all default to the value `otlp`. There is no value other than `otlp` that would satisfy `OTEL_EXPORTER` because zipkin doesn't do metrics, etc. It does affect this sort of use case where it is a good idea to change the normal default of `otlp` to `none` - we would need to discuss in spec if that means the `OTEL_EXPORTER` variable would be good to have for this use case.

Reply:
Thanks for you patience @anuraaga , I hope I understood well your point, please check the updated code, it's greatly simplified.